Source/WebCore/rendering/svg/RenderSVGPath.cpp

```diff
@@ -52,15 +52,22 @@ RenderSVGPath::~RenderSVGPath() = default;
 void RenderSVGPath::updateShapeFromElement()
 {
     RenderSVGShape::updateShapeFromElement();
+    processMarkerPositions();
     updateZeroLengthSubpaths();
 
-    m_strokeBoundingBox = calculateUpdatedStrokeBoundingBox();
+    m_strokeBoundingBox = calculateUpdatedStrokeBoundingBox(m_strokeBoundingBox);
```

Comment:
I think the life cycle of the `m_path` and the flags `m_needsShapeUpdate` and `m_needsBoundariesUpdate` became more unclear and fragile with this patch. I think after creating `m_path` and updating it from the element, `m_needsShapeUpdate` should be set to false. And I think `m_strokeBoundingBox`, `m_fillBoundingBox` and `m_repaintBoundingBox` should be calculated in something called `updateBoundariesBoxes()` and after calling it, `m_needsBoundariesUpdate` should be set to false.

Reply:
Please have a look at https://github.com/WebKit/WebKit/pull/18490. This crash is happening because `m_path` is null even after running the layout. This shows that this code is fragile and there are cases that we are not aware of and they are hard to find.

Reply:
I see #18490, and as the bug is showing, this is not having `m_path` even without this patch's change. (so existing bug). I'll comment on that issue about it.

The reason of that bug is not directly related to this change since,

1. The bug is anyway happening without the change.
2. The bug is that LegacyRenderSVGPath is not having a `m_path`. Since that class always has a path when `updateShapeFromElement` happens, this means that it is not doing layout before, while `isRenderingDisabled` code in LegacyRenderSVGPath is requiring that.
3. Stacktrace of #18490 is showing that `LegacyRenderSVGRoot::layout` is called for the container, and it is calling `SVGRenderSupport::layoutChildren`. This means that the bug is happening because of inconsistency between `isRenderingDisabled` condition and layout-skipping condition in `SVGRenderSupport::layoutChildren`. If we align them, then it is fine. If we skipped the layout (this is what is happening in that bug), then layout related condition is not usable, including `m_path`. Using layout dependent condition in `isRenderingDisabled` is not right.

I think that calling `updateShapeFromElement` from various places makes lifecycle tracking harder, as the above issue is showing the crash. The better direction is that making it much more explicit and clear (Always update from `updateShapeFromElement` calls by layout)

So,
> This crash is happening because m_path is null even after running the layout. 

This is not the reason of the crash in #18490. The root cause of the bug is, layout is not properly happening for that LegacyRenderSVGPath, and/or not using the right condition in `isRenderingDisabled`.
It is skipping layout because of the condition in `SVGRenderSupport::layoutChildren`, and not using the same condition in `isRenderingDisabled`. That's the issue. If we skip the layout with a condition, then we need to check the same condition again. `isRenderingDisabled` works only if the layout is once called. If the layout is skipped, using it is not the right way.